src/main/java/com/ning/http/client/providers/grizzly/ConnectionManager.java

```diff
@@ -219,10 +219,11 @@ static boolean isKeepAlive(final Connection connection) {
         return !IS_NOT_KEEP_ALIVE.isSet(connection);
     }
     
-    private static String getPartitionId(Request request,
+    private static String getPartitionId(InetAddress overrideAddress, Request request,
             ProxyServer proxyServer) {
-        return request.getConnectionPoolPartitioning()
-                .getPartitionKey(request.getUri(), proxyServer).toString();
+        return overrideAddress != null ? overrideAddress.toString() + "_" : "" + 
```

Comment:
is it supposed to be
(overrideAddress != null ? overrideAddress.toString() + "_" : "") +
           request.getConnectionPoolPartitioning()
           .getPartitionKey(request.getUri(), proxyServer).toString();

?